dolphin/drivers/dell_emc/vmax/client.py

```diff
@@ -123,3 +123,58 @@ def list_pools(self):
             LOG.error("Failed to get pool metrics from VMAX: {}".format(err))
             raise exception.StorageBackendException(
                 reason='Failed to get pool metrics from VMAX')
+
+    def list_volumes(self, storage_id):
+
+        try:
+            # List all volumes
+            volumes = self.conn.provisioning.get_volume_list(filters={'data_volume': 'false'})
+
+            volume_list = []
+            for volume in volumes:
+                # Get volume details
+                vol = self.conn.provisioning.get_volume(volume)
+
+                total_cap = vol['cap_mb'] * units.Mi
```

Comment:
After getting vol, we are accessing some attributes of vol. Are we sure they are always there or else we can add some check for the existence of necessary parameters to avoid exception

Reply:
We need this fields from backend. Else it is exception.

Reply:
Ok if we are getting them under all circumstances, its ok. Otherwise better to handle and throw some error message to upper layer

Reply:
We can't guaranty availability of all fields.. based on config and model it would change.. better to have a check on optional fields.. mandatory fields like cap_mb can go for exception..

Reply:
Currently, we raise exception for unavailability of each field. Which is safer option, until we know what are optional fields. Also, currently we do not know, which additional model we needs to support.

Reply:
fine.